pkg/koordlet/util/meminfo.go

```diff
@@ -114,13 +142,95 @@ func readMemInfo(path string) (*MemInfo, error) {
 	return &info, nil
 }
 
+func GetMemInfo() (*MemInfo, error) {
+	memInfoPath := system.GetProcFilePath(system.ProcMemInfoName)
+	memInfo, err := readMemInfo(memInfoPath, false)
+	if err != nil {
+		return nil, err
+	}
+	return memInfo, nil
+}
+
 // GetMemInfoUsageKB returns the node's memory usage quantity (kB)
 func GetMemInfoUsageKB() (int64, error) {
 	meminfoPath := system.GetProcFilePath(system.ProcMemInfoName)
-	memInfo, err := readMemInfo(meminfoPath)
+	memInfo, err := readMemInfo(meminfoPath, false)
 	if err != nil {
 		return 0, err
 	}
 	usage := int64(memInfo.MemTotal - memInfo.MemAvailable)
 	return usage, nil
 }
+
+type NUMAInfo struct {
```

Comment:
is socket id can be identified?

Reply:
I'm afraid not. There is no info about the socket and core relations under the numa info directory. If we want to look up the socket ID of the NUMA node, we need to check the cpu info instead.